vnext/include/ReactUWP/InstanceFactory.h

```diff
@@ -18,6 +18,7 @@ struct IXamlRootView;
 struct IReactInstanceCreator {
   virtual std::shared_ptr<IReactInstance> getInstance() = 0;
   virtual void markAsNeedsReload() = 0;
+  virtual void persistUseWebDebugger(bool useWebDebugger) = 0;
```

Comment:
so this means as part of integrating RNW each client app right now has to implement this new method?  Not sure if there is a way to internally persist this and have clients app request or get notified of what the new devsettings/ReactInstanceSettings are?

Reply:
We have some work to do in this area...  Today the app is required to create a ReactInstanceSettings and pass it into IReactInstance::Start().  Turning remote debugging on/off requires us to tear down and restart, so I can't persist anything internally, it will just get stomped over by the app when we ask it to Start a new instance.

I sort of feel like this is a wrong-way dependency.  We could change this so that the framework is the true "owner" of the settings, and perhaps provide APIs to set/get them.  Note that Micah/Jon are in the process of overhauling all of the APIs between the framework and host, so this could just be a short term problem.

Reply:
+@vmoroz , this is basically what reactnativehost does internally.  -- Have we had any thoughts on moving RNH out into RNW? -- Otherwise RNH is going to get re-implemented out in RNW.

And does it even work for win32?

Reply:
@ahimberg, @acoates-ms, @vmoroz, @kmelmon I've created a cpp/winrt project that wraps react.uwp and exposes an API for app code that is more similar to what's provided by the existing C# RN.  It has a ReactInstanceSettings as a protected property on ReactNativeHost.  if an app would sub-class ReactNativeHost and return their own settings object if they didn't want the default.  you can see an example at https://github.com/micahl/react-native-windows/blob/microsoftreact/vnext/Playground/PlaygroundCS/MainReactNativeHost.cs.

Reply:
@acoates-ms, current RNH works for Win32 and it is already started being used by some projects. Based on our offline discussion, I am going to start looking into moving the RNH into RNW.
@micahl, thank you for sharing your playground solution! I am a little bit unsure about using the inheritance. I believe the general .NET framework guidelines are against using sub-classing as a way to extend framework functionality. It is better to provide all settings as input arguments. E.g. the .Net Dictionary is generally never sub-classed, but rather provides explicit API to tweak its behavior. Also, as @aeulitz works on the ABI safe-API based on WInRT API, we should check with him if the inheritance is going to work when C# consumes the DLL created in C++.